ingestion/src/metadata/utils/connections.py

```diff
@@ -539,3 +561,39 @@ def _(connection: PowerBiClient) -> None:
         raise SourceConnectionException(
             f"Unknown error connecting with {connection} - {err}."
         )
+
+
+@test_connection.register
+def _(connection: S3Client) -> None:
+    """
+    Test that we can connect to the source using the given aws resource
+    :param engine: boto service resource to test
+    :return: None or raise an exception if we cannot connect
+    """
+    from botocore.client import ClientError
+
+    try:
+        connection.client.list_buckets()
```

Comment:
this might not be the best way to test this here. We are requesting a bucket_name, so the role would just need permissions of getting object and listing objects, but not about listing buckets. There might be another way to create and test the connection